p2p/src/peer_manager/peerdb.rs

```diff
@@ -92,8 +121,11 @@ pub struct PeerDb<T: NetworkingService> {
     /// Pending connections
     pending: HashMap<T::Address, T::PeerId>,
 
-    /// Banned peers
-    banned: HashSet<T::PeerId>,
+    /// Banned addresses along with the duration of the ban.
+    ///
+    /// The duration represents the `UNIX_EPOCH + duration` time point, so the ban should end
+    /// when `current_time > ban_duration`.
+    banned: BTreeMap<IpAddr, Duration>,
```

Comment:
Can `T::Address` be stored here? As far as I get `IpAddr` is only required to unify different types of addresses. But maybe storing polymorpic type is enough?

Reply:
Also, there can be another associated type for a "BannableAddress", if this isn't appropriate.

Reply:
. Currently we have three address types:
- `std::net::SocketAddr`
- `libp2p::MultiAddr`
- A channel address (`u64`)

The last one isn't important for now. A socket address consists of an IP address and a port and I think that we want to ban the "whole" address and not a specific address and port combination. The situation is even more trickier with libp2p because the address can be the following: `/ip4/7.7.7.7/tcp/4242/p2p/QmYyQSo1c1Ym7orWxLYvCrM2EmxFTANf8wXmmE7DWjhx5N` and once again I think that we are only interested in the IP address here.